### PR TITLE
Updates ORCA geometry parsing.

### PR DESCRIPTION
**Pull request by jevandezande**
Simplifies geometry parsing by combining single point and optimization geometry parsing. Adds parsing for atommasses and a test case.

Notes
-----
- Apparently different programs have different weights for atoms, necessitating large error bars.

Questions
---------
- [x] Is there a way to run a test only for a set of parsers (e.g. with a decorator like `runForParser` akin to `skipForParser`)?
- [x] I had to use the ugly test `self.assertTrue(numpy.isclose(atommass, H, 3) or numpy.isclose(atommass, C, 1))` in order to pass the regressions as some regressions have the atoms ordered differently. Could the regressions be changed to have the same starting geometries and atom order?

Status
------
- [ ] Regression tests need to be merged, PR cclib/cclib-data#60

### CODE REVIEW

**Comment by berquist** [comment]
> Apparently different programs have different weights for atoms, necessitating large error bars.

I think `atommasses` was originally intended to hold the isotopic masses used in frequency calculations. This may explain part of it (where the masses were originally taken from), plus the printing precision. Due to the low precision it's hard to tell exactly what ORCA is printing here, but I assume it's most common isotope, which is the most important thing.

> Is there a way to run a test only for a set of parsers (e.g. with a decorator like runForParser akin to skipForParser)?

Maybe something like this (invert the run/skip condition from `skipForParser`):

```python
def runForParser(parser, msg):
    """Return a decorator that runs the test for only the specified parser."""
    def testdecorator(testfunc):
        def testwrapper(self, *args, **kwargs):
            if self.logfile.logname == parser:
                testfunc(self, *args, **kwargs)
            else:
                self.skipTest(msg)
        return testwrapper
    return testdecorator
```

We probably don't want to do this; by forcing opt-out as the default behavior, it's more clear when reading the tests what specific problems are with specific parsers. This would make it harder to go back and update tests so that they run for all parsers.

> I had to use the ugly test self.assertTrue(numpy.isclose(atommass, H, 3) or numpy.isclose(atommass, C, 1)) in order to pass the regressions as some regressions have the atoms ordered differently. Could the regressions be changed to have the same starting geometries and atom order?

Yes, the geometries are inconsistent between programs, plus they have [other problems](https://github.com/cclib/cclib/issues/136). Fixing this in general will be a difficult undertaking because no one has access to ADF, GAMESS-UK, or Jaguar right now.

For now, it would be better to test both elements (so use `and` rather than `or`) by searching for the first instance of each.

**Comment by langner** [comment]
Yes, `atommasses` was meant for the common isotope.

I agree with Eric, and would prefer to skip with skipping parsers. If you want, you could create a `skipForPrasers` decorator to save a few lines.

Although we make an effort to standardize unit test logfiles, there will be no such effort for old unit tests in regressions. However, we can skip single tests or make tweaks to old unit tests in regressions as needed.

**Comment by langner** [comment]
> Yes, atommasses was meant for the common isotope.

Correction - it was to hold whatever the program prints. So, you might request some rare isotope if you're studying its effect on something.

File: test/data/testGeoOpt.py [hunk 1]
```diff
@@ -123,6 +123,18 @@ def testgeovalues_atomcoords(self):
         msg = "len(atomcoords) is %d but len(geovalues) is %d" % (count_coords, count_geovalues)
         self.assertEquals(count_geovalues, count_coords, msg)
 
+    @skipForParser("GAMESSUK", "Not implemented.")
```
Comment (langner) [inline test/data/testGeoOpt.py]: So, do we need this test given we do an equivalent test in `testSP`? Are the masses formatted any differently for geometry optimization?

Also, please take the time time to enable this test for ORCA in `testSP`. Thanks!

File: test/data/testSP.py [hunk 2]
```diff
@@ -101,15 +101,17 @@ def testatombasis(self):
...
     @skipForLogfile('Psi/basicPsi4.0b5', 'atommasses not implemented yet')
     @skipForParser('QChem', 'atommasses not implemented yet')
     def testatommasses(self):
         """Do the atom masses sum up to the molecular mass?"""
         mm = 1000*sum(self.data.atommasses)
-        msg = "Molecule mass: %f not %f +- %fmD" % (mm, self.molecularmass, self.mass_precision)
-        self.assertAlmostEquals(mm, self.molecularmass, delta=self.mass_precision, msg=msg)
+        molecularmass = self.molecularmass
+        if self.logfile.logname == 'ORCA':
+            molecularmass = 130190.0
```
Comment (berquist) [inline test/data/testSP.py]: @langner Would it be better to create a `ORCASPTest` class? There is precedent: see `Psi3SPTest`, where only the total energy is different.

Comment (langner) [inline test/data/testSP.py]: Sure... that's the convention we've been using and this fits the use case.

**Pull request merged**